src/main/java/org/elasticsearch/common/xcontent/XContentFactory.java

```diff
@@ -208,14 +208,16 @@ public static XContentType xContentType(byte[] data) {
      * Guesses the content type based on the provided input stream.
      */
     public static XContentType xContentType(InputStream si) throws IOException {
-        int first = si.read();
-        if (first == -1) {
+        int iFirst = si.read();
```

Comment:
can't you simply cast to byte directly? the signed extension is preserved though